roles/backpack_kube/tasks/main.yml

```diff
@@ -0,0 +1,89 @@
+---
+- name: Create Backpack Namespace
+  k8s: 
+    name: backpack
+    api_version: v1
+    kind: Namespace
+    state: present
+    kubeconfig: "{{ kube_config }}"
+
+- name: Launch backpack daemonset
+  k8s:
+    state: present
+    definition: "{{ lookup('file', 'roles/backpack_kube/files/backpack_set.yaml') }}"
+    kubeconfig: "{{ kube_config }}"
+  register: my_daemon_set
+
+- name: Sleep 5 to allow process time
+  pause:
+    seconds: 5
+
+- name: Wait until all pods exit creating state
+  k8s_facts:
+    kind: Pod
+    kubeconfig: "{{ kube_config }}"
+    label_selectors:
+      - name = backpack
+  register: pod_list
+  until: "'Creating' not in pod_list | json_query('resources[].status.phase')"
+
+
+- name: Wait for all pods to contain BACKPACK_DONE
+  shell: "/usr/local/bin/kubectl --kubeconfig={{ kube_config }} -n backpack logs -l name=backpack | grep BACKPACK_DONE | wc -l"
+  register: done_count
+  until: "done_count.stdout|int == pod_list.resources|length"
+  retries: 60
+  delay: 10
+
+- name: Create temporary directory
+  file:
+    path: "{{ tmp_dir}}"
+    state: directory
+    mode: "u=rw,g=rwx,o=rwx"
+
+- name: Dump logs to temporary directory
+  shell: "/usr/local/bin/kubectl --kubeconfig={{ kube_config }} -n backpack logs {{ item }} | grep -v BACKPACK_DONE > {{ tmp_dir }}/{{ item }}"
+  with_items:
+    - "{{ pod_list | json_query('resources[].metadata.name') }}"
+
+- name: Replace pod name with node name
+  replace:
+    path: "{{ tmp_dir }}/{{ item.metadata.name }}"
+    regexp: "{{ item.metadata.name }}"
+    replace: "{{ item.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution | json_query('nodeSelectorTerms[].matchFields[].values')|first|first }}"
+  with_items:
+      - "{{ pod_list.resources }}"
+
+- name: Replace leading localhost with node name
```

Comment:
Can you please clarify a little bit on why we're replacing pod name with node name as well as localhost with node name?

Reply:
If we don't replace localhost with something useful we won't have an easy way to tell what node it was run on. For example, prior to implementing the replacement there would be no reference to minikube.